### PR TITLE
Writing flow: fix vertical arrow nav in table (and generally grid)

### LINKED ISSUE #14675: Arrow key navigation in table block
Spun off from https://github.com/WordPress/gutenberg/pull/14462#issuecomment-476165700

When the caret is at the table block:

- the arrow down should move to the cell below instead of to the right cell
- the arrow up key should move the caret to the cell above instead of the left cell

### PR DESCRIPTION
**Pull request by ellatrix**
## Description

Fixes #14675.

When using the up and down arrow keys, and there are tabbable elements placed in a row (e.g. table), the focus visually moves horizontally, which feels wrong.

The solution is to exclude tabbable elements that are in the same row.

table-nav
## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/e2e-tests/specs/editor/various/writing-flow.test.js [hunk 1]
```diff
@@ -80,25 +80,7 @@ describe( 'Writing Flow', () => {
 		activeBlockName = await getActiveBlockName();
 		expect( activeBlockName ).toBe( 'core/column' );
 		await page.keyboard.press( 'ArrowUp' );
-		activeBlockName = await getActiveBlockName();
```
Comment (ellatrix) [inline packages/e2e-tests/specs/editor/various/writing-flow.test.js:83]: This is expected. Arrow up shouldn't go through all the previous columns.

**Comment by ellatrix** [comment]
@talldan May be of interest to you since you've also looked at navigation for grids. Hopefully this is a good general solution. We can tweak it if we have more cases.

**Comment by talldan** [comment]
Hey @ellatrix That's pretty cool, much simpler than the solution I had!

I did some testing and mostly it seems to work well, though I noticed that sometimes when moving in to the last cell in a table in Firefox, I'm unable to move back out again using the left or up arrow keys.

**Comment by ellatrix** [comment]
It seems to work fine for me. It does scroll when I press arrow up, but the focus is correct. Do you have more specific step to reproduce?

table-nav-ff

**Pull request merged**